File: docs-examples/example-java/src/test/java/io/micronaut/nats/docs/quickstart/QuickstartSpec.java
package io.micronaut.nats.docs.quickstart;

import io.micronaut.context.annotation.Property;
import io.micronaut.nats.testcontainers.Nats;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import io.micronaut.test.support.TestPropertyProvider;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;

import java.util.Map;

import static java.util.concurrent.TimeUnit.SECONDS;
import static org.awaitility.Awaitility.await;

@MicronautTest
@Property(name = "spec.name", value = "QuickstartSpec")
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class QuickstartSpec implements TestPropertyProvider {

    @Test
    void testProductClientAndListener(ProductClient productClient, ProductListener productListener) {

// tag::producer[]
productClient.send("quickstart".getBytes());
// end::producer[]

        await().atMost(60, SECONDS).until(() ->
                productListener.messageLengths.size() == 1 &&
                productListener.messageLengths.get(0).equals("quickstart")
        );
    }

    @Override
    public Map<String, String> getProperties() {
        return Nats.getProperties();
    }
}

File: docs-examples/example-python/src/test/python/micronaut/nats/docs/quickstart/QuickstartSpec.py
from time import sleep
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context.annotation import Property
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from .ProductClient import ProductClient
from .ProductListener import ProductListener


@Property(name="spec.name", value="QuickstartSpec")
@MicronautTest(environments=["nats"])
class QuickstartSpec:
    product_client: Annotated[ProductClient, Inject]
    product_listener: Annotated[ProductListener, Inject]

    @Test
    def test_product_client_and_listener(self):
        # tag::producer[]
        self.product_client.send("quickstart".encode())
        # end::producer[]

        for _ in range(600):
            if len(self.product_listener.message_lengths) == 1 and self.product_listener.message_lengths[0] == "quickstart":
                break
            sleep(0.1)
        assert self.product_listener.message_lengths == ["quickstart"]
